client/verta/verta/tracking/entities/_deployable_entity.py

```diff
@@ -189,8 +190,21 @@ def download_artifact(self, key, download_to_path):
             Absolute path where artifact was downloaded to. Matches `download_to_path`.
 
         """
+        download_to_path = os.path.abspath(download_to_path)
+        artifact = self._get_artifact_msg(key)
+        if artifact.upload_completed:
+            pathlib2.Path(download_to_path).parent.mkdir(parents=True, exist_ok=True)
+            # TODO: clean up empty parent dirs if something later fails
+            return self._internal_download_artifact(key, artifact, download_to_path)
+        else:
+            raise ValueError("artifact upload incomplete; download not possible")
+
+
+    @abc.abstractmethod
+    def _internal_download_artifact(self, key, artifact, download_to_path):
```

Comment:
I'd like to request this be named either `_download_artifact_internal()` (in the vein of our existing `_create_proto_internal()`s) or `_download_artifact_inner()` (in the vein of our existing `_from_dict_inner()`s).

I'd push for the former, since "internal" seems to have been the more intuitive choice ("inner" was mine 😬)